src/transports/smart_protocol.c

```diff
@@ -317,27 +317,34 @@ static int fetch_setup_walk(git_revwalk **out, git_repository *repo)
 static int wait_while_ack(gitno_buffer *buf)
 {
 	int error;
-	git_pkt_ack *pkt = NULL;
+	git_pkt *pkt = NULL;
+	git_pkt_ack *ack = NULL;
 
 	while (1) {
-		git__free(pkt);
+		if (pkt) {
```

Comment:
We usually do not check whether a pointer is set before freeing it. But in this case it is necessary, as `git_pkt_free` doesn't check for `NULL` pointers itself